Framework/API/src/MatrixWorkspace.cpp

```diff
@@ -72,11 +72,6 @@ MatrixWorkspace::MatrixWorkspace(const MatrixWorkspace &other)
   m_isCommonBinsFlag = other.m_isCommonBinsFlag;
   m_masks = other.m_masks;
   // TODO: Do we need to init m_monitorWorkspace?
-
```

Comment:
Yes: The copy constructor is using `copyExperimentInfoFrom`, which is using `setInstrument`, which as you point out is not doing the copy directly.